cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -405,9 +409,22 @@ func StartControllers(s *options.CMServer, kubeClient *client.Client, kubeconfig
 		time.Sleep(wait.Jitter(s.ControllerStartInterval.Duration, ControllerStartJitter))
 	}
 
-	go volume.NewAttachDetachController(clientset.NewForConfigOrDie(restclient.AddUserAgent(kubeconfig, "attachdetach-controller")), podInformer, nodeInformer, ResyncPeriod(s)()).
-		Run(wait.NeverStop)
-	time.Sleep(wait.Jitter(s.ControllerStartInterval.Duration, ControllerStartJitter))
+	attachDetachController, attachDetachControllerErr :=
+		volume.NewAttachDetachController(
+			clientset.NewForConfigOrDie(restclient.AddUserAgent(kubeconfig, "attachdetach-controller")),
+			podInformer,
+			nodeInformer,
+			pvcInformer,
+			pvInformer,
+			ResyncPeriod(s)(),
+			cloud,
+			ProbeAttachableVolumePlugins(s.VolumeConfiguration.FlexVolumePluginDir))
```

Comment:
This feels a bit awkward.. perhaps pass the whole VolumeConfig struct and fish out the settings needed for each plugin in the constructor. It is only FlexVolume now but may expand in the future